### PR TITLE
fix: validate RepositoryVulnerabilityAlert to getVulnerability

### PR DESCRIPTION
**Pull request by sugarshin**
Hi, there. This pull-request is related to #11911 and solves it.

Now, I facing this problem too. The problem of #11911 is caused by the Renovate does not null-check for it. As [GitHub documentation](https://docs.github.com/en/graphql/reference/objects#repositoryvulnerabilityalert), `securityVulnerability` is nullable. but, I think fixing it to be greatly affected, such as proper TypeScript type definitions.

~~And, I think [this line](https://github.com/renovatebot/renovate/blob/8e09d295678c68d9a0b6dc471a693c5fddb03b10/lib/platform/github/index.ts#L1728) of logging with error and exiting with non-zero status code related to the above problem is not appropriate because the try-catch here is a process to parse the vulnerability alerts for just logging. It means not affected to `getVulnerabilityAlerts()` return values. And is not related to the original execution behavior of the Renovate.~~

~~So I think `logger.warn` is better.~~

## Changes:

Validate for `securityVulnerability` to skip processing parse RepositoryVulnerabilityAlert entity for logger.

## Context:

- https://github.com/renovatebot/renovate/issues/11911

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by JamieMagee** [comment]
Can you open an issue to discuss this issue? Using `logger.error` doesn't make Renovate exit with a non-zero exit code. You can argue that it should be `warning` because it's a recoverable error, but I don't think that this change alone will solve the issue you're seeing.

**Comment by sugarshin** [comment]
@JamieMagee Thank you for your reply.

> logger.error doesn't make Renovate exit with a non-zero exit code

No, `logger.error` does make Renovate exit with a non-zero status code. Could you check following?

https://github.com/renovatebot/renovate/blob/8e09d295678c68d9a0b6dc471a693c5fddb03b10/lib/workers/global/index.ts#L144-L151

**Comment by JamieMagee** [comment]
My mistake! Okay, then this may well be the fix. Can you still give me a bit more information about your setup? The original issue you linked was running on GHES and for the npm package `hoek`. What sort of setup is causing this error for you?

**Comment by sugarshin** [comment]
@JamieMagee Thank you! We construct Renovate bot with self-hosting.

The global config is the following:

```js
module.exports = {
  endpoint: 'https://<my-company-ghe-domain>/api/v3/',
  token: process.env.GHE_TOKEN,
  platform: 'github',
  logLevel: 'debug',
  onboardingConfig: {
    extends: ['config:base'],
  },
  cacheDir: __dirname + '/.renovate_cache',
  dryRun: !!process.env.RENOVATE_DRYRUN,
  autodiscover: false,
  packageRules: [
    {
      matchDatasources: ['docker'],
      registryUrls: ['<masked>']
    }
  ],
  repositories: [
    // repositories
  ],
};
```

The config of the repository where this problem occurs is the following:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "ignorePaths": [
    "**/node_modules/**"
  ],
  "packageRules": [
    {
      "matchDepTypes": ["major"],
      "enabled": false
    },
    {
      "matchDepTypes": ["devDependencies"],
      "addLabels": ["devDependencies"]
    }
  ],
  "labels": ["renovate"]
}
```

I attached a screenshot that error log:

ss

**Comment by sugarshin** [comment]
And, I executed with `LOG_LEVEL=trace` to test. The log content of [this line](https://github.com/renovatebot/renovate/blob/8e09d295678c68d9a0b6dc471a693c5fddb03b10/lib/platform/github/index.ts#L1707):

```json
{
    "name": "renovate",
    "hostname": "<masked>",
    "pid": 709,
    "level": 10,
    "logContext": "MjvOWnPA6xLTJre-KECJU",
    "repository": "<masked>",
    "alerts": [
        {
            "dismissReason": null,
            "vulnerableManifestFilename": "package-lock.json",
            "vulnerableManifestPath": "package-lock.json",
            "vulnerableRequirements": "= 2.2.1",
            "securityAdvisory": null,
            "securityVulnerability": null
        },
        ...
```

As you can see, `"securityVulnerability"` is `null`.

**Comment by rarkins** [comment]
Seems like a faulty alert?

I'd prefer to fix the validation (eg so that we can still process the non faulty ones) than simply lower the log severity

**Comment by sugarshin** [comment]
@rarkins Thank you for your comment.

> I'd prefer to fix the validation (eg so that we can still process the non faulty ones)

Ok, I agree with you. I will fix it later.

**Comment by JamieMagee** [comment]
@sugarshin Can you give me the package name for the malformed vulnerability alert? Maybe I can follow-up internally about it.

**Comment by sugarshin** [comment]
@JamieMagee Thank you. I couldn't determine which package, but I have several candidates:

```
app-root-path@2.2.1
ansi-styles@2.2.1
cli-width@2.2.1
readdirp@2.2.1
md5@2.2.1
underscore.string@2.2.1
tapable@2.2.1
```

Could check it?

**Comment by sugarshin** [comment]
@rarkins I fixed it, so could you check it?

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -1706,6 +1706,12 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
       if (alerts.length) {
         logger.trace({ alerts }, 'GitHub vulnerability details');
         for (const alert of alerts) {
+          if (!alert.securityVulnerability) {
```
Comment (viceice) [inline lib/platform/github/index.ts:1709]:
```suggestion
          if (alert.securityVulnerability === null) {
```
We should be explicit here. `undefined` should still be an error.

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

File: lib/platform/github/index.ts (hunk 1 above)
Comment (sugarshin) [reply lib/platform/github/index.ts]: @viceice Thank you for your suggestion. I commit it.

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by viceice** [comment]
I think this will also fix https://github.com/renovatebot/renovate/issues/13796

**Pull request merged**